### PR TITLE
[WIP] initial CI support

### PR DESCRIPTION
**Pull request by funtimeerror**
This PR is still a WIP.

Adding support for travis-ci so that we can make sure the code wont have any syntax errors, and also runs the PSR2 code checks (I disabled the checks mentioned in the contributing guide), it will also check any PR's submitted as well.

@chriskonnertz After setting up an account on https://travis-ci.org any pushes to the branch with the .travis.yml file will be auto-built and checked. This service is free for open source code :)

I had to disable the ExampleTest for checking for status 200 as I havnt been able to get a status response back yet from the CI engine correctly (keeps coming back as a 404)

### CODE REVIEW

File: tests/Feature/ExampleTest.php [hunk 1]
```diff
@@ -16,6 +16,6 @@ public function testBasicTest()
     {
         $response = $this->get('/');
 
-        $response->assertStatus(200);
+        //$response->assertStatus(200);
```
Comment (chriskonnertz) [inline tests/Feature/ExampleTest.php:19]: That's fine, this test comes from Laravel and really is just an example, there is no need to actually run it or to do any assertions.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  - '7.1'
+  - '7.2'
+  - '7.3'
+  
+install:
+  - php composer.phar update --no-suggest
+  - php composer.phar install --no-suggest
+  
+before_script:
+  - if find . -name "*.php" ! -path "./vendor/*" -print0 | xargs -0 -n 1 -P 8 php -l | grep -v "No syntax errors detected"; then exit 1; fi
```
Comment (chriskonnertz) [inline .travis.yml:13]: Fascinating. I recently read the PHP documentation about the CLI options but I did not really take notice of the lint option... This is pretty nice, using it in combination with a shell script to check all PHP files for errors. I like it!

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+dist: xenial
+language: php
+php:
+  - '7.1'
```
Comment (chriskonnertz) [inline .travis.yml:4]: We should also test for 7.0, because this will be the minimum requirement for v3.0.

**Review by chriskonnertz — COMMENTED** [review]
Hello,

cool. I wrote a few comments. :)

**Comment by chriskonnertz** [comment]
Merged it into a feature branch so Travis can do its job. I expect a lot of PSR-2 errors.

**Comment by chriskonnertz** [comment]
Ok, Travis did it's job. Yep, a lot of PSR-2 violations :D 

=> https://travis-ci.org/Contentify/Contentify/jobs/510923472

We should check if we can make the code sniffer ignore this error:

> The closing brace for the class must go on the next line after the body

**Comment by chriskonnertz** [comment]
Merged the temp branch into the `3.0-dev` branch with these changes:
https://github.com/Contentify/Contentify/pull/530/commits/febcd120431f7ae96b12368df3c5b3914a4707bd
So PHP 7.0 is in and for now it makes sense to ignore the PSR-2-checks, because too many debatable errors are thrown.

**Comment by funtimeerror** [comment]
@chrisKonnertz yea im still working on the PSR2 violations. I should have a PR created for that tonight after work (~7hrs from now)

its gonna be a LARGE PR :)

**Pull request merged**